lib/flags.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+class Flags {
+  static from(...args) {
+    const values = args;
+    class FlagsClass {
+      static from(...args) {
+        return new FlagsClass(...args);
+      }
+
+      constructor(...args) {
+        this.values = new Map(values.map(v => [v, 0]));
+        args.forEach(arg => this.set(arg));
+      }
+
+      has(key) {
+        return this.values.has(key);
+      }
+
+      get(key) {
+        return this.values.get(key);
+      }
+
+      set(key) {
+        if (this.values.has(key)) this.values.set(key, 1);
+      }
+
+      unset(key) {
+        if (this.values.has(key)) this.values.set(key, 0);
+      }
+
+      switch(key) {
+        if (this.values.has(key))
```

Comment:
There must not be such checks. All bits are 'unset' by default.

Reply:
```js
  const Numeri = Flags.from('Uno', 'Due', 'Tre', 'Quatro');
  const num = Numeri.from('Due', 'Tre');
  num.switch('Cinque');
```
The 3rd line shouldn't work, because this key wasn't defined at `Flags.from`.

Reply:
The third line is incorrect usage, that must not be allowed.
Upon creation (`Flags.from`), you define all of the allowed values.

Reply:
I agree, but there should be some form of check. I propose to throw error if there is no such defined key. Wdyt?

Reply:
Just FYI: `Enum` doesn't need such checks since it doesn't provide interface to change values and is not intended to do this.

Reply:
Well, it does :smile: https://github.com/metarhia/common/blob/master/lib/enum.js#L14 it cannot change values but you still have to do something when a user asked for an enum instance of a key that doesn't exist.

Reply:
Yes, this is a nice proposal :+1: I'll add `NaN` as default value for `get` if key doesn't exist.

Reply:
I'm not sure, where do you add `NaN`? This is supposed to be BitPacked field I don't think there is any place to put that `NaN`?